### DISCUSSION TITLE
options data in imports table

### DISCUSSION THREAD

**Discussion by smiliyas** [filamentphp/filament#13700] [Ideas]
@danharrin, what are your thoughts on storing the `$columnMap` and `$options` data in the `imports` table?

This could be a breaking change in v3, but we could implement it in v4. If you give me the green light, I’m ready to make a PR for this. By doing so, we could remove these properties from `import` events.

**Comment by danharrin** [comment]
What would be the benefits?

**Reply by smiliyas** [reply]
Basically it will be helpful to track the history of the imports by some additional conditions.

Let's assume, I am importing records which belong to a tenant. By having tenant id in options data, I can easily filter the imports which belong to a particular tenant.

**Reply by danharrin** [reply]
I would accept a PR for the feature in v4, as long as you also include a section in the upgrade script that publishes a migration to add the new columns IF the old migration was published
